cli/cli.go

```diff
@@ -357,14 +364,175 @@ func getMigrateSubcommand(subcmd, usage string) cli.Command {
+		return
+	}
+	return
+}
+
+func createDbFromConfig(config *util.Config) (db.DB, error) {
+	dbType := config.GetString("database/type", "sqlite3")
+	dbConnection := config.GetString("database/connection", "")
+	dbConn := sql.NewDB()
+	err := dbConn.Connect(dbType, dbConnection, db.DefaultMaxOpenConn)
+	if err != nil {
+		return nil, err
+	}
+	return dbConn, nil
+}
+
+func createIdentityFromConfig(config *util.Config) (middleware.IdentityService, error) {
+	if config.GetBool("keystone/use_keystone", false) {
+		if config.GetBool("keystone/fake", false) {
+			log.Info("Debug Mode with Fake Keystone Server")
+			return &middleware.FakeIdentity{}, nil
+
+		} else {
+			log.Info("Keystone backend server configured")
+			keystoneIdentity, err := cloud.NewKeystoneIdentity(
+				config.GetString("keystone/auth_url", "http://localhost:35357/v3"),
+				config.GetString("keystone/user_name", "admin"),
+				config.GetString("keystone/password", "password"),
+				config.GetString("keystone/domain_name", "Default"),
+				config.GetString("keystone/tenant_name", "admin"),
+				config.GetString("keystone/version", ""))
+			if err != nil {
+				log.Fatal("Failed to create keystone identity service, err: %s", err)
+			}
+			return keystoneIdentity, nil
+		}
+	}
+	return nil, fmt.Errorf("No identity service defined in config")
+}
+
+func getMigrateSubcommandWithPostMigrateEvent(subcmd, usage string) cli.Command {
+	const (
+		POST_MIGRATION_EVENT_TIMEOUT_FLAG = "post-migration-event-timeout"
+		CONFIG_FILE_FLAG                  = "config-file"
+		EMIT_POST_MIGRATION_EVENT_FLAG    = "emit-post-migration-event"
+		POST_MIGRATION_EVENT              = "post-migration"
+	)
+	return cli.Command{
+		Name:  subcmd,
+		Usage: usage,
+		Flags: []cli.Flag{
+			cli.StringFlag{Name: CONFIG_FILE_FLAG, Value: defaultConfigFile, Usage: "Server config File"},
+			cli.BoolFlag{Name: EMIT_POST_MIGRATION_EVENT_FLAG, Usage: "Enable if post-migration event should be emited to modified schema extensions"},
+			cli.DurationFlag{Name: POST_MIGRATION_EVENT_TIMEOUT_FLAG, Value: time.Second * 30, Usage: "Maximum duration of post-migration event"},
+		},
+		Action: func(context *cli.Context) {
+			configFile := context.String(CONFIG_FILE_FLAG)
+			if migration.LoadConfig(configFile) != nil {
+				return
+			}
+			config := util.GetConfig()
+
+			migration.Run(subcmd, context.Args())
+
+			emitEvent := context.Bool(EMIT_POST_MIGRATION_EVENT_FLAG)
+			if !emitEvent {
+				return
+			}
+
+			modifiedSchemas := migration.GetModifiedSchemas()
+
+			if len(modifiedSchemas) == 0 {
+				log.Info("No modified schemas, skipping post-migration event")
+				return
+			}
+
+			log.Debug("Modified schemas: %s", modifiedSchemas)
+
+			schemaFiles := config.GetStringList("schemas", nil)
+			if schemaFiles == nil {
+				log.Fatal("No schema specified in configuraion")
+			}
+
+			manager := schema.GetManager()
+			if err := manager.LoadSchemasFromFiles(schemaFiles...); err != nil {
```

Comment:
is it possible that a post-migration event handler in one schema triggers an event (eg. update) on a different schema? if so, I think we have to load all schemas here.